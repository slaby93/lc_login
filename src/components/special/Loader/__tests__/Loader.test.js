import React from 'react';
import { shallow } from 'enzyme';
import LoaderStyled, { Loader } from './../Loader';

describe('<Loader />', () => {
  test('CSS', () => {
    const mountedComponent = shallow(<LoaderStyled />);
    expect(mountedComponent).toMatchSnapshot();
  });

  test('HTML', () => {
    const mountedComponent = shallow(<Loader
      className="test"
    />);
    expect(mountedComponent).toMatchSnapshot();
  });

  describe('props', () => {
    test('container', () => {
      const mountedComponent = shallow(<LoaderStyled container />);
      expect(mountedComponent).toMatchSnapshot();
    });
  });
});
