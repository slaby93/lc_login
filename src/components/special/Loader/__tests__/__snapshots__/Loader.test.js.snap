// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> CSS 1`] = `
.c0 .spinner-box {
  width: 50px;
  height: 50px;
  position: relative;
}

.c0 .spinner-box:before {
  content: "";
  display: block;
  width: 100%;
  height: 100%;
  border-radius: 50%;
  border: solid 5px;
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
}

.c0 .spinner-axis {
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  -webkit-animation: spin 0.8s steps(100,end) infinite;
  -webkit-animation: spin 0.8s steps(100,end) infinite;
  animation: spin 0.8s steps(100,end) infinite;
}

.c0 .spinner {
  width: 50%;
  height: 50%;
  overflow: hidden;
}

.c0 .spinner-inner {
  width: 200%;
  height: 200%;
  border-radius: 100%;
  border: solid 5px;
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
}

<Loader
  className="c0"
/>
`;

exports[`<Loader /> HTML 1`] = `
<div
  className="test"
>
  <div
    className="spinner-box"
  >
    <div
      className="spinner-axis"
    >
      <div
        className="spinner"
      >
        <div
          className="spinner-inner"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Loader /> props container 1`] = `
.c0 {
  background-color: #f1f1f169;
  width: 100%;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  left: 0;
  top: 0;
  z-index: 1;
}

.c0 .spinner-box {
  width: 50px;
  height: 50px;
  position: relative;
}

.c0 .spinner-box:before {
  content: "";
  display: block;
  width: 100%;
  height: 100%;
  border-radius: 50%;
  border: solid 5px;
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
}

.c0 .spinner-axis {
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  -webkit-animation: spin 0.8s steps(100,end) infinite;
  -webkit-animation: spin 0.8s steps(100,end) infinite;
  animation: spin 0.8s steps(100,end) infinite;
}

.c0 .spinner {
  width: 50%;
  height: 50%;
  overflow: hidden;
}

.c0 .spinner-inner {
  width: 200%;
  height: 200%;
  border-radius: 100%;
  border: solid 5px;
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
}

.c0 .spinner-box {
  top: 25%;
}

<Loader
  className="c0"
  container={true}
/>
`;
